alacritty/src/input.rs

```diff
@@ -153,12 +159,38 @@ impl<T: EventListener> Execute<T> for Action {
                 ctx.scroll(Scroll::Bottom);
                 ctx.write_to_pty(s.clone().into_bytes())
             },
-            Action::Command(ref program) => {
-                let args = program.args();
-                let program = program.program();
+            Action::Command(ref cmd) => {
+                let args = cmd.args().to_owned();
+                let input = cmd.input();
+                let program = cmd.program().to_string();
+                let esc_seqs = cmd.esc_seqs();
+                let term = ctx.terminal();
                 trace!("Running command {} with args {:?}", program, args);
 
-                start_daemon(program, args);
+                let input = input.map(|i| match i {
+                    CommandInput::VisibleText => {
+                        term.grid_collector_only_visible(CollectorMode::Sync, esc_seqs)
+                    },
+                    CommandInput::AllText => {
+                        term.grid_collector(CollectorMode::from(*term.screen_lines()), esc_seqs)
+                    },
+                });
+
+                let term = ctx.terminal_ptr().clone();
+                thread::spawn_named("command", move || {
+                    let input = input.map(|mut collector| {
+                        let mut done = false;
+                        while !done {
+                            let term = term.lock();
+                            done = collector.execute(term.grid());
+                            mem::drop(term);
+                        }
+
+                        collector.finish()
+                    });
```

Comment:
Do you spawn your stuff in a thread here because it's so slow?

Reply:
No. I spawn it in a thread because it allows me to make it asynchronous.